src/popup/pages/Retip.vue

```diff
@@ -33,149 +33,196 @@
-        if (this.formModel.selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
-          await this.$store.dispatch('fungibleTokens/createOrChangeAllowance',
-            [this.formModel.selectedAsset.contractId, this.formModel.amount]);
-          retipResponse = await this.tippingV2.methods.retip_token(
-            +this.tip.id.split('_')[0],
-            this.formModel.selectedAsset.contractId,
+        if (formModel.value.selectedAsset.contractId !== AETERNITY_CONTRACT_ID) {
+          await root.$store.dispatch('fungibleTokens/createOrChangeAllowance',
+            [formModel.value.selectedAsset.contractId, formModel.value.amount]);
+          retipResponse = await tippingV2.value.methods.retip_token(
+            +tip.value.id.split('_')[0],
+            formModel.value.selectedAsset.contractId,
             amount,
             {
               waitMined: false,
             },
           );
         } else {
-          retipResponse = await this.tippingContract.methods.retip(+this.tip.id.split('_')[0], {
+          retipResponse = await tippingContract.value.methods.retip(+tip.value.id.split('_')[0], {
             amount,
             waitMined: false,
           });
         }
-        this.$store.dispatch('addPendingTransaction', {
+        root.$store.dispatch('addPendingTransaction', {
           hash: retipResponse.hash,
           amount,
-          tipUrl: this.tip.url,
+          tipUrl: tip.value.url,
           tx: {
-            callerId: this.account.address,
-            contractId: this.tippingContract.deployInfo.address,
+            callerId: account.value.address,
+            contractId: tippingContract.value.deployInfo.address,
             type: SCHEMA.TX_TYPE.contractCall,
             function: 'retip',
-            selectedTokenId: this.formModel.selectedAsset?.contractId,
+            selectedTokenId: formModel.value.selectedAsset?.contractId,
           },
         });
-        this.openCallbackOrGoHome(true);
-      } catch (e) {
-        this.$store.dispatch('modals/open', {
+        openCallbackOrGoHome(true);
+      } catch (e:any) {
+        root.$store.dispatch('modals/open', {
           name: 'default',
-          title: this.$t('modals.transaction-failed.msg'),
+          title: root.$t('modals.transaction-failed.msg'),
           icon: 'critical',
         });
-        e.payload = this.tip;
+        e.payload = tip.value;
         throw e;
       } finally {
-        this.loading = false;
+        loading.value = false;
       }
-    },
+    };
+
+    const handleAssetChange = (selectedAsset: any) => {
+      formModel.value.selectedAsset = selectedAsset;
+    };
+
+    onMounted(async () => {
+      loading.value = true;
+      formModel.value.selectedAsset = getAeternityToken.value({
+        tokenBalance: balance.value,
+        balanceCurrency: balanceCurrency.value,
+      });
+      const tipId = root.$route.query.id;
+      if (!tipId) throw new Error('"id" param is missed');
```

Comment:
I think the message should be
```suggestion
      if (!tipId) throw new Error('"id" param is missing');
```